test/topology_custom/test_view_build_status.py

```diff
@@ -489,3 +489,63 @@ async def rows_migrated():
         return (len(result) == 3) or None
 
     await wait_for(rows_migrated, time.time() + 60)
+
+# Reproduces scylladb/scylladb#20754
+@pytest.mark.asyncio
+@skip_mode('release', 'error injection is not supported in release mode')
+async def test_migration_on_existing_raft_topology(request, manager: ManagerClient):
+    cfg = {
+        "error_injections_at_startup": [
+            {
+                "name": "suppress_features",
+                "value": "VIEW_BUILD_STATUS_ON_GROUP0"
+            },
+            "skip_vb_v2_version_mut"
+        ]
+    }
+    servers = await manager.servers_add(3, config=cfg)
+
+    logging.info("Waiting until driver connects to every server")
+    cql, hosts = await manager.get_ready_cql(servers)
+
+    await create_keyspace(cql)
+    await create_table(cql)
+    await create_mv(cql, "vt1")
+
+    # Verify we're using v1 now
+    v = await get_view_builder_version(cql)
+    assert v == 1
+
+    async def _view_build_finished():
+        result = await cql.run_async("SELECT * FROM system_distributed.view_build_status")
+        return len(result) == 3
+    await wait_for(_view_build_finished, time.time() + 10, period=.5)
+
+    result = await cql.run_async("SELECT * FROM system.view_build_status_v2")
+    assert len(result) == 0
+
+    # Enable suppressed `VIEW_BUILD_STATUS_ON_GROUP0` cluster feature
+    for srv in servers:
+        await manager.server_stop_gracefully(srv.server_id)
+        await manager.server_update_config(srv.server_id, "error_injections_at_startup", [])
+        await manager.server_start(srv.server_id)
+        await wait_for_cql_and_get_hosts(manager.get_cql(), servers, time.time() + 60)
+
+    logging.info("Waiting until upgrade finishes")
+    await asyncio.gather(*(wait_until_topology_upgrade_finishes(manager, h.address, time.time() + 60) for h in hosts))
```

Comment:
The cluster starts in topology on raft mode, no need to wait until topology upgrade finishes.

The "Checking migrated data in system" part actually waits until view builder migration finishes. Perhaps the text should be changed slightly to mention that you wait and don't just check - if you were just checking then the code should fail if the check fails, but here you use `wait_for` which will keep retrying the check.